### PR TITLE
Fix 443 release notes

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Correcting Delta Lake connector release note (... **`timestamp(6)`**  ...).

## Additional context and related issues

https://github.com/trinodb/trino/pull/21178#discussion_r1536565429

https://github.com/trinodb/trino/pull/20814

### CODE REVIEW

File: docs/src/main/sphinx/release/release-443.md [hunk 1]
```diff
@@ -43,4 +43,4 @@
   connector plugins. ({issue}`21114`)
 
 
-Thanks again for presenting at Trino Summit last year .. sending you a reminder to register for Trino Fest in June ..https://trino.io/blog/2024/02/20/announcing-trino-fest-2024
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:46]: Unrelated change

Comment (mosabua) [reply docs/src/main/sphinx/release/release-443.md:46]: Here is a PR with just that removal

**Review by wendigo — APPROVED** [review]

**Review by mosabua — APPROVED** [review]
Good to go
